### ISSUE TITLE
How should I handle different routing keys?

### ISSUE THREAD

**Issue by herenow** [jondot/sneakers#143]
Hi,

I would like to ask what is the suggested way to handle messages with different routing keys, for example:

Imagine I have a service, that handles the following messages:

```
payload, routing_key: "user.reset.password"
payload, routing_key: "user.reset.email"
```

In my mind, in Sneakers I have two options, spawn 2 workers that each consume "user.reset.password" and "user.reset.email", or I could have 1 worker, that consumes "user.reset.*" with `work_with_params` instead of `work` and acts upon `delivery_info.routing_key`. Currently, I'm not a fan of either solution, since the former introduces overhead,
 and IMO makes handling concurrency harder; the latter IMO is not a good pattern, it adds unnecessary complexity to the worker.

I'm new to rabbitmq, and I may have my thinking all wrong :)

Thanks.

**Comment by taelor** [comment]
@herenow If it helps you feel any better, I do something 90% similar to your second method. I've created a module interface for my ActiveRecord obejcts, and they produce message keys that are "class_name.action", where action is create, update or destroy (to mimic the callbacks in AR). I send over a json representation of that object as the paylod.

Now when sneakers receives the message, I can figure out the Class is, and what action was performed. Combined with the json, you get a lot of information to use and play around for little effort. 

I've actually added more metadata to that key, since we are a multi-tenant application. I've thought about even adding the "id" to the routing key. "user.create.3" or something.
